dash/CMakeLists.txt

```diff
@@ -255,6 +258,7 @@ foreach (dart_variant ${DART_IMPLEMENTATIONS_LIST})
   set (DART_LIBRARY "dart-${dart_variant}")
   set (DASH_LIBRARY "dash-${dart_variant}")
   set (VARIANT_ADDITIONAL_COMPILE_FLAGS ${ADDITIONAL_COMPILE_FLAGS_STR})
+  set (VARIANT_ADDITIONAL_COMPILE_FLAGS ${CXX_OMP_FLAG})
```

Comment:
@devreal I just noticed that there is a little problem with this line.
It drops the `ADDITINAL_COMPILE_FLAGS` by overwriting the variable.

Those flags include:
```
-DDASH
-DDASH_ENABLE_DEFAULT_INDEX_TYPE_LONG
-DDASH_ENABLE_THREADSUPPORT
-DDASH_ENABLE_UNIFIED_MEMORY_MODEL
-DDASH_ENABLE_TEST_LOGGING
-DDASH_ENABLE_PAPI
-DDASH_ENABLE_HWLOC
-DDASH_ENABLE_NUMA
-DDASH_ENABLE_HDF5
-DDASH_ENV_HOST_SYSTEM_ID='default'
```
I will provide a hotfix for this.

Reply:
Gnarf, you're right. Please provide the fix in a PR. And thanks for taking care of it.